src/Compilers/Test/Core/Assert/ConditionalFactAttribute.cs

```diff
@@ -159,6 +159,7 @@ public static class ExecutionConditionUtil
         public static bool IsWindowsDesktop => IsWindows && IsDesktop;
         public static bool IsMonoDesktop => Type.GetType("Mono.Runtime") != null;
         public static bool IsMono => MonoHelpers.IsRunningOnMono();
+        public static bool IsNewMono => Type.GetType("Mono.RuntimeStructs") != null;
```

Comment:
IsMono - looks for Desktop/legacy mono . 
Figured this out when I was trying to skip couple of test cases on mono environment using ClrOnly() which uses IsRunningOnMono(). 
ClrOnly() doesn't hold good for mono platforms . 

Hence to figure out if it is running on mono runtime or not implemented another check IsNewMono which uses IsUsingMonoRuntime() .

Reply:
Is there anything more information I can provide you with ?

Reply:
I'm still not clear though what the new mono is. I'm very familiar with desktop mono. What is the new variant of mono that you're trying to account for here?

Reply:
There is the Mono runtime which lives in dotnet/runtime and is used for WASM and other scenarios. It appears that `Mono.RuntimeStructs` lives in both mono/mono and dotnet/runtime. `Mono.Runtime` seems to only be in mono/mono.

Reply:
Yes, this test is about checking for running on .NET with the Mono runtime (as opposed to the CoreCLR runtime).  There is also a similar check already present in roslyn here: https://github.com/dotnet/roslyn/blob/54fd790788b03c256f736c917fc368daf0a643db/src/Compilers/Core/Portable/InternalUtilities/PlatformInformation.cs#L44

Maybe this can be re-used.  Or at least, the same terminology can be used - I think something like `IsUsingMonoRuntime` (or `IsMonoRuntime` like here https://github.com/dotnet/runtime/blob/b5833d2cd22f474485f74696609f6a6c3617cd15/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs#L30) would be preferable over `IsNewMono`.